src/server/auth/server/api_server.go

```diff
@@ -124,35 +130,122 @@ func NewAuthServer(env Env, public, requireNoncriticalServers, watchesEnabled bo
 }
 
 func (a *AuthServer) EnvBootstrap(ctx context.Context) error {
-	if a.env.Config.AuthRootToken == "" {
+	if !a.env.Config.ActivateAuth {
 		return nil
 	}
 	a.env.Logger.Info("Started to configure auth server via environment")
 	if err := func() error {
-		if _, err := a.Activate(ctx, &auth.ActivateRequest{
-			RootToken: a.env.Config.AuthRootToken,
-		}); err != nil {
-			if !errors.As(err, auth.ErrAlreadyActivated) {
+		ctx = internalauth.AsInternalUser(ctx, internalUser)
+		// handle auth activation
+		if a.env.Config.AuthRootToken != "" {
+			if _, err := a.Activate(ctx, &auth.ActivateRequest{
+				RootToken: a.env.Config.AuthRootToken,
+			}); err != nil {
+				if !errors.As(err, auth.ErrAlreadyActivated) {
+					return err
```

Comment:
Let's wrap this, and the (existing) error on line 151, and the PPS/PFS ActivateAuth calls.